tests/testthat/test-variable-catalog.R

```diff
@@ -101,12 +100,77 @@ with_mock_crunch({
                 type=c("numeric", "categorical", "categorical", "multiple_response")
             )))
     })
+
+    test_that("Variable Catagory as.data.frame method", {
+        expect_equal(as.data.frame(varcat[1:3]),
+            data.frame(
+                alias = c("birthyr", "gender", "location"),
+                name  = c("Birth Year", "Gender", "Categorical Location"),
+                type  = c("numeric", "categorical", "categorical"),
+                stringsAsFactors = FALSE
+            ))
+    })
+    test_that("as.data.frame method returns all fields when keys = 'all'", {
+        expect_identical(
+            names(as.data.frame(varcat, keys = "all")),
+            c("name", "discarded", "alias", "type", "id", "description",
+                "notes", "subvariables", "subvariables_catalog", "resolution",
+                "rollup_resolution")
+        )
+    })
+    test_that("As.data.frame method errors correctly", {
+        expect_error(as.data.frame(varcat[1:3], keys = "Not a field at all"),
+            paste(dQuote("Not a field at all"), "is an invalid key for catalogs of class VariableCatalog.")
+        )
+        expect_error(as.data.frame(varcat[1:3], keys = c("banana", "fooey")),
+            paste(
+                serialPaste(dQuote(c("banana", "fooey"))),
+                "are invalid keys for catalogs of class VariableCatalog.")
+        )
+        expect_error(as.data.frame(varcat[1:3], keys = c("name", "fooey")),
+            paste(dQuote("fooey"), "is an invalid key for catalogs of class VariableCatalog.")
+        )
+    })
+
+    test_that("entryToDF works", {
+      entry <- list(
+          alias = "birthyr",
+          name = "Birth Year",
+          type = "numeric")
+      list_entry <- list(
+          name = "mymrset",
+          discarded = FALSE,
+          alias = "mymrset",
+          type = "multiple_response",
+          id = "949d2dc7e7a24e6090cc88bb92e1d2fb",
+          description = "Please select all that apply", notes = "",
+          subvariables = list(
+              "mymrset/subvariables/subvar2/",
+              "mymrset/subvariables/subvar1/",
+              "mymrset/subvariables/subvar3/"
+              ),
+          subvariables_catalog = "mymrset/subvariables/"
+          )
+
+      expected_df <- data.frame(alias = "birthyr",
+                                name = "Birth Year",
+                                type = "numeric",
+                                stringsAsFactors = FALSE)
+      expect_identical(entryToDF(entry, c("subvariables", "subvariables_catalog")),
+        expected_df)
+      expect_identical(entryToDF(entry, c("subvariables", "not_a_variable")),
+        expected_df)
```

Comment:
Aren't these both testing that when you give listcols that don't exist you still get what you expect back?

Additionally, it would be good to test what the format looks like when you expect there to be list columns to be present.

Reply:
One is asking for two variables which exists, but isn't a list-column, one includes a variable which doesn't exist.